pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -757,6 +757,23 @@ private static Expression compileFunctionExpression(SqlBasicCall functionNode) {
     }
   }
 
+  private static final Map<String, FilterKind> CANONICAL_NAME_TO_FILTER_KIND_MAP = new HashMap<String, FilterKind>() {{
+    for (FilterKind filterKind : FilterKind.values()) {
+      put(StringUtils.remove(filterKind.name(), '_'), filterKind);
+    }
+  }};
```

Comment:
This is widely considered an antipattern because it creates a new class for each map this is done for. Doing this in a static block or using a factory method doesn’t add much code but doesn’t increase the number of class files, classes to load etc.

Reply:
Good point, didn't know that.
Here is an article explaining why this is an anti-pattern: https://blog.jooq.org/dont-be-clever-the-double-curly-braces-anti-pattern